service/terraform-plugin/src/main/java/com/solace/maas/ep/event/management/agent/plugin/terraform/manager/TerraformManager.java

```diff
@@ -52,65 +47,78 @@ public void execute(CommandRequest request, Command command, Map<String, String>
+            TerraformUtils.setCommandError(command, e);
         }
     }
 
-    private static List<String> setupTerraformClient(TerraformClient terraformClient, Path configPath) {
-        terraformClient.setWorkingDirectory(configPath.toFile());
-        List<String> output = new ArrayList<>();
-
-        // Write each terraform to the output list so that it can be processed later
-        // Also write the output to the main log to be streamed back to EP
-        terraformClient.setOutputListener(tfLog -> {
-            output.add(tfLog);
-            logToConsole(tfLog);
-        });
-        return output;
+    private static void setEnvVarsFromParameters(Command command, Map<String, String> envVars) {
+        if (command.getParameters() != null && command.getParameters().containsKey("environment") &&
+                command.getParameters().get("environment") instanceof Map) {
+            Map<String, String> environmentMap = (Map<String, String>) command.getParameters().get("environment");
+            environmentMap.keySet().forEach(key -> envVars.put("TF_VAR_" + key, environmentMap.get(key)));
+        }
     }
 
-    @SneakyThrows
-    private static void logToConsole(String tfLog) {
-
-        String logMessage = String.format("Terraform output: %s", tfLog);
+    private List<String> executeTerraformCommand(Command command, Map<String, String> envVars, Path configPath,
+                                                 TerraformClient terraformClient) throws IOException, InterruptedException, ExecutionException {
+        String commandVerb = command.getCommand();
 
-        Map<String, Object> logMop = objectMapper.readValue(tfLog, Map.class);
-        String logLevel = (String) logMop.get("@level");
-        switch (logLevel) {
-            case "trace" -> log.trace(logMessage);
-            case "debug" -> log.debug(logMessage);
-            case "info" -> log.info(logMessage);
-            case "warn" -> log.warn(logMessage);
-            case "error" -> log.error(logMessage);
-            default -> log.error("cannot map the logLevel properly for tfLog {}", tfLog);
+        Consumer<String> logToConsole = (tfLog) -> logToConsole(tfLog);
+        if ("import".equals(commandVerb)) {
+            logToConsole = (log) -> {
+            };
         }
-    }
+        List<String> logOutput = TerraformUtils.setupTerraformClient(terraformClient, configPath, logToConsole);
 
-    private static String executeTerraformCommand(Command command, Map<String, String> envVars, Path configPath, TerraformClient terraformClient) throws IOException, InterruptedException, ExecutionException {
-        String commandVerb = command.getCommand();
         switch (commandVerb) {
+            case "import" -> importCommand(envVars, configPath, terraformClient, logOutput);
             case "apply" -> {
-                writeHclToFile(command, configPath);
-                Boolean planSuccessful = terraformClient.plan(envVars).get();
```

Comment:
Calling terraform `plan` before each `apply` was not adding value and only adds to the amount of time it takes to execute and doubles the number of logs. Removed it streamline the apply.